File: src/Python/260.SingleNumberIII.py

class Solution:
    def singleNumber(self, nums):
        sign = 0
        for num in nums:
            sign = sign ^ num
        sign &= -sign
        re = [0] * 2
        for num in nums:
            if num & sign == sign:
                re[0] ^= num
            else:
                re[1] ^= num
        return re

    
if __name__ == "__main__":
    solution = Solution()
    print(solution.singleNumber([1, 2, 1, 3, 2, 5]))

File: src/CSharp/260.SingleNumberIII.cs

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;

namespace PeefyLeetCode.SingleNumberIII
{
    public class Solution
    {
        public int[] SingleNumber(int[] nums)
        {
            int sign = 0;
            for (int i = 0; i < nums.Length; i++)
            {
                sign ^= nums[i];
            }
            sign &= -sign;
            int[] result = new int[2];
            for (int i = 0; i < nums.Length; i++)
            {
                if ((nums[i] & sign) == sign)
                    result[0] ^= nums[i];
                else
                    result[1] ^= nums[i]; ;
            }
            return result;
        }
    }
}